src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -1,27 +1,64 @@
-import React from 'react';
+import React, { useRef, useState } from 'react';
 import PropTypes from 'prop-types';
 import CustomFieldInputText from '../custom-field-input-text/custom-field-input-text.jsx';
 import Icon from '../icon/icon.jsx';
 import iconCaretDown from '../../svgs/icon-caret-down.svg';
 import iconCaretDownDisabled from '../../svgs/icon-caret-down-disabled.svg';
-import styles from '../custom-field-input-text/custom-field-input-text.css';
+import styles from './custom-field-input-single-choice.css';
+import Listbox from '../listbox/listbox.jsx';
+import ListOption from '../list-option/list-option.jsx';
 
 export default function CustomFieldInputSingleChoice(props) {
+  const [showOptions, setShowOptions] = useState(false);
+
+  const refs = props.choices.map(() => useRef());
+  const caretIcon = (<Icon
+    className={styles['input-icon']}
+    name={props.readOnly ? iconCaretDownDisabled.id : iconCaretDown.id}
+    fill="skip"
+  />);
+
+  function onClick() {
+    setShowOptions(true);
+  }
+
+  function onKeyUp(event) {
+    if (event.key === 'Enter') {
+      setShowOptions(true);
+    } else if (event.key === 'Escape') {
+      setShowOptions(false);
+    }
+  }
+
   return (
-    <CustomFieldInputText
-      icon={<Icon className={styles['input-icon']} name={props.readOnly ? iconCaretDownDisabled.id : iconCaretDown.id} fill="skip" />}
-      id={props.id}
-      label={props.label}
-      placeholder={props.placeholder}
-      readOnly={props.readOnly}
-      required={props.required}
-      value={props.value}
-    />
+    <div className={styles.container}>
+      <CustomFieldInputText
+        icon={caretIcon}
+        id={props.id}
+        label={props.label}
+        onClick={onClick}
+        onKeyUp={onKeyUp}
+        placeholder={props.placeholder}
+        readOnly={props.readOnly}
+        required={props.required}
+        value={props.value}
+      />
+      { showOptions && (
+        <Listbox
```

Comment:
Rendering an empty listbox is showing up weirdly. Let's ask design what they want. We might want it to not render at all.